samples/v2/component_with_optional_inputs.yaml

```diff
@@ -0,0 +1,99 @@
+# PIPELINE DEFINITION
+# Name: v2-component-optional-input
+components:
```

Comment:
We should also add the python code from which each of these yaml specs have been generated.

Reply:
Actually the file name would explain this -> `component_with_optional_inputs.py` is the python code for `component_with_optional_inputs.yaml`

Reply:
But I don't see the harm in adding a comment in each of these `.yaml` files again